### PR TITLE
feat: add protocol schemes to index and filtering by protocols to TM list

### PR DESCRIPTION
**Pull request by alexbrdn**

### CODE REVIEW

File: internal/model/model.go [hunk 1]
```diff
@@ -32,6 +38,84 @@ type Version struct {
...
+		}
+	}
+	return protos
+}
+
+func extractProtocol(uri string) string {
+	if uri == "" {
+		return ""
+	}
+	u, err := url.Parse(uri)
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply internal/model/model.go:116]: Good point. Check out the solution I came up with for this. What do you think?

File: internal/model/model.go [hunk 2]
```diff
@@ -32,6 +38,84 @@ type Version struct {
...
+	}
+
+	protos, _ := collectProtocols(data)
+	(&tm).setProtocols(protos)
+	return &tm, nil
+}
+
+func (tm *ThingModel) setProtocols(protos []string) {
+	slices.Sort(protos)
+	slices.Compact(protos)
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply internal/model/model.go]: removed

File: internal/model/model.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (enricoschuetz) [reply internal/model/model.go:116]: I think your solution is good and sufficient for most cases.

File: internal/model/model.go [hunk 3]
```diff
@@ -32,6 +38,88 @@ type Version struct {
...
+	}
+
+	protos, _ := collectProtocols(data)
+	(&tm).setProtocols(protos)
+	return &tm, nil
+}
+
+func (tm *ThingModel) setProtocols(protos []string) {
+	tm.protocols = protos
+}
```
Comment (enricoschuetz) [inline internal/model/model.go:55]: As the function is only called in `ParseThingModel`, I think the protocols on the TM (this one line of code) can be directly set in `ParseThingModel`.

**Pull request merged**